kubetools_client/config.py

```diff
@@ -52,11 +52,35 @@ def load_kubetools_config(
         dev (bool): filter config items by dev mode
     '''
 
-    possible_files = (
+    possible_filenames = (
         'kubetools.yml',
         'kubetools.yaml',
     )
 
+    if directory:
+        possible_files = [
+            path.join(directory, filename)
+            for filename in possible_filenames
+        ]
+    else:
+        # Look in this directory
+        directory = getcwd()
+        possible_files = [
+            path.join(directory, filename)
+            for filename in possible_filenames
+        ]
+
+        # Attempt parent directories back up to root
+        while True:
```

Comment:
Hmmm only thought here is that this will continue to read outside of any repo. I wonder if we only want to read up to a certain point? E.g. my repos are typically laid out as
```
~
    - source
        - repos
            - project_1
                - some_sub_dir
            - project_2
                - some_other_sub_dir
```

Suppose I'm in `~/source/repos/project_1/some_sub_dir/`, then it feels like we should only check as far as `~/source/repos/project_1`, and not look for `~/kubetools.yml`, `~/source/kubetools.yml` or `~/source/repos/kubetools.yml`. Not sure how best to implement this - we could do `git rev-parse --show-toplevel`, but that assumes being in a git repository. Or maybe it's not an issue and we shouldn't worry. Thoughts?

Reply:
Agreed, but I can't see any clean solution to the problem here

Reply:
Yeah this is really hard/impossible to solve I think - can't see any way of knowing the correct point to stop looking "up" a directory. Is this better or worse than the current directory only lookup?

Reply:
I'd still say this is better - I find myself far more often wanting to use `ktd` from within a subdirectory than creating `kubetools.yml` outside a repo, and will happily allow it to check for some files that don't exist if it allows that functionality.

Really the only case I can see this being an issue is with git submodules (where both repos have a `kubetools.yml`), but that's a real edge case and if we're doing that we should expect a few gotchas like this.